### PR TITLE
[Bindings] Append Direction and Route as Built-in Metadata Properties.

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description

Modifies the `(c *ComponentMetadata) AppendBuiltin()` to add the `direction` and `route` builtin properties.

## Issue reference
PR will close: #2944

Related issues:
    https://github.com/dapr/dapr/pull/6509 
   
    https://github.com/dapr/docs/issues/3541

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:27]: In Go, convention is to use PascalCase for constants.

Comment (ItalyPaleAle) [reply .build-tools/pkg/metadataschema/validators.go:27]: Or camelCase if not exported

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:167]:
```suggestion
					Description: "Indicates the direction of the binding component.",
```

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:171]: This link isn't working for me - it doesn't point to the section specified here

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:182]:
```suggestion
						Example:     "`/customroute`",
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [reply .build-tools/pkg/metadataschema/validators.go:171]: I will merge this PR once the docs PR is merged, so docs maintainer confirm the URL

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:165]: Do you think we should include this option only if the binding is both input and output? Otherwise it can be confusing.

Comment (yaron2) [reply .build-tools/pkg/metadataschema/validators.go:165]: This property was specifically designed to let users specify that a binding is not both input and output

Comment (ItalyPaleAle) [reply .build-tools/pkg/metadataschema/validators.go:165]: The metadata property "direction" should only be included if the component offers input binding capabilities. A few lines higher you use the `c.Binding.Input` and `c.Binding.Output` properties that indicate if the component is an input and/or output binding.

- If the component is both input and output (`c.Binding.Input && c.Binding.Output`):
   - Include the "direction" metadata property with allowed values (as **enum** type): `input`,`output`,`input,output` (no space)
- If the component is **just** input (`c.Binding.Input`):
   - Include the "direction" metadata property with only 1 allowed value (as **enum** type): `input` (adding this will make Dapr not perform the initial OPTIONS request)
- If the component is **just** output (`c.Binding.Output`):
   - Metadata property "direction" is not present

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:181]:
```suggestion
						Example:     `"/custom-path"`,
```

File: .build-tools/pkg/metadataschema/validators.go
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:167]:
```suggestion
						Example:     `"`+direction+`"`,
```

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**
